OSCompatibilityLayer.h

```diff
@@ -47,67 +45,65 @@ typedef int errno_t;
-
-	std::string convertUTF8ToASCII(const std::string& UTF8);
-	std::string convertUTF8To8859_15(const std::string& UTF8);
-	std::string convertUTF8ToWin1252(const std::string& UTF8);
-	std::string convertUTF16ToUTF8(const std::wstring& UTF16);
-	std::string convert8859_15ToASCII(const std::string& input);
-	std::string convert8859_15ToUTF8(const std::string& input);
-	std::wstring convert8859_15ToUTF16(const std::string& input);
-	std::string convertWin1252ToASCII(const std::string& Win1252);
-	std::string convertWin1252ToUTF8(const std::string& Win1252);
-	std::string convertWin1250ToUTF8(const std::string& Win1252);
-	std::wstring convertWin1252ToUTF16(const std::string& Win1252);
-	std::wstring convertWin1250ToUTF16(const std::string& Win1250);
-	std::wstring convertUTF8ToUTF16(const std::string& UTF8);
-	std::string convertWin1251ToUTF8(const std::string& Win1251);
-	std::string convertUTF8toWin1251(const std::string& UTF8);
-
-	// converts a string in the system dependent wchar_t encoding to UTF-8
-	std::string convertToUTF8(const std::wstring &input);
-
-	//converts an UTF8 path to the system dependent filesystem path encoding
-	std::string normalizeUTF8Path(const std::string &utf_8_path);
+// Creates a new folder corresponding to the given path.
+// Returns true on success or if the folder already exists.
+// Returns false and logs a warning on failure.
+bool TryCreateFolder(const std::string& path);
+std::string GetCurrentDirectory();
+// Adds all files (just the file name) in the specified folder to the given collection.
+void GetAllFilesInFolder(const std::string& path, std::set<std::string>& fileNames);
+void GetAllSubfolders(const std::string& path, std::set<std::string>& subfolders);
+
+// For the specified folder and all subfolders, adds all files (just the subdirectory and file name)
+// to the given collection.
+void GetAllFilesInFolderRecursive(const std::string& path, std::set<std::string>& filenames);
+// Copies the file specified by sourcePath as destPath.
+// Returns true on success.
+// Returns false and logs a warning on failure.
+bool TryCopyFile(const std::string& sourcePath, const std::string& destPath);
+bool CopyFolder(const std::string& sourceFolder, const std::string& destFolder);
+bool RenameFolder(const std::string& sourceFolder, const std::string& destFolder);
+// Returns true if the specified file exists (and is a file rather than a folder).
+bool DoesFileExist(const std::string& path);
+// Returns true if the specified folder exists (and is a folder rather than a file).
+bool DoesFolderExist(const std::string& path);
+
+void WriteToConsole(LogLevel level, const std::string& logMessage);
+
+// Returns a formatted string describing the last error on the WinAPI.
+std::string GetLastErrorString();
+
+bool DeleteFolder(const std::string& folder);
+
+
+std::optional<std::string> GetFileFromTag(const std::string& directoryPath, const std::string& tag);
```

Comment:
This seems to not really fit the overall theme here. As well, it seems unused in any of the current converters. Consider deleting it, or at the very least deprecating it.